File: cpp/status.cpp
#include "status.h"

Status::Status(std::string label) {  
    this->label = label;
}

std::string Status::getLabel() const{
    return this->label;
}

const Status Status::ACTIVE = Status("Active");
const Status Status::HYBERNATED = Status("Hybernated");
const Status Status::STOPPED = Status("Stopped");
const Status Status::TERMINATED = Status("Terminated");

void Status::setLabel(std::string label){
    this->label = label;
}

File: js/Status.js
class Status {
    #label;
    
    static ACTIVE = new Status('Active');
    static HYBERNATED = new Status('Hybernated');
    static STOPPED = new Status('Stopped');
    static TERMINATED = new Status('Terminated');

    constructor(label) {
        this.#label = label;
    }
    
    getLabel(){
        return this.#label;
    }
    
    setLabel(label){
        this.#label = label;
    }
}

module.exports = Status;
